### PR TITLE
Add Holt Winters Model for time series

### PR DESCRIPTION
**Pull request by SuvarshaChennareddy**
Related to #2668

### CODE REVIEW

**Review by zoq — COMMENTED** [review]
Let me know if you need some help with the test case.

**Comment by SuvarshaChennareddy** [comment]
@zoq Thank you! I guess I do need help..I don't really know why the tests are failing in several checks.

File: src/mlpack/tests/main_tests/holt_winters_test.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ */
+    TEST_CASE_METHOD(HoltWintersTestFixture, "DimensionsTest",
+        "[HoltWintersMainTest][BindingTests]")
+{
+    arma::mat dataset(9, 3, arma::fill::randu);
+    SetInputParam("input", std::move(dataset));
+    SetInputParam("period", 5);
+    SetInputParam("numberOfForecasts", 1);
+    //Throws an error beacuse the dataset must have atleast 2*period elements
+    REQUIRE_THROWS(RUN_BINDING());
```
Comment (dnabanita7) [inline src/mlpack/tests/main_tests/holt_winters_test.cpp:37]: For checking any error, you can use ``REQUIRE_THROWS_AS`` or  ``REQUIRE_THROWS_WITH`` or any of the other forms defined in ``catch.hpp``.

File: src/mlpack/tests/main_tests/holt_winters_test.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ * making sure an error is thrown when dataset has zero elements.
+ */
+TEST_CASE_METHOD(HoltWintersTestFixture, "EmptyDataTest",
+    "[HoltWintersMainTest][BindingTests]")
+{
+    arma::mat dataset(0, 3, arma::fill::randu);
+    SetInputParam("input", std::move(dataset));
+    SetInputParam("period", 1);
+    SetInputParam("numberOfForecasts", 1);
+    REQUIRE_THROWS(RUN_BINDING());
```
Comment (dnabanita7) [inline src/mlpack/tests/main_tests/holt_winters_test.cpp:51]: Same as above

**Comment by dnabanita7** [comment]
I think the issue is bindings to various languages are not getting created as you can see any of the language bindings build is not available.

**Comment by SuvarshaChennareddy** [comment]
Keep open

**Comment by SuvarshaChennareddy** [comment]
> I think the issue is bindings to various languages are not getting created as you can see any of the language bindings build is not available.

Yes I can see that, but I don't really know what I missed? Do you have any idea?

File: src/mlpack/tests/main_tests/holt_winters_test.cpp (hunk 2 above)
Comment (SuvarshaChennareddy) [reply src/mlpack/tests/main_tests/holt_winters_test.cpp:51]: Alright thank you for the suggestions!

**Pull request closed without merging**